test/integration_tests/integrationtest_scanner_api.cpp

```diff
@@ -0,0 +1,149 @@
+// Copyright (c) 2020 Pilz GmbH & Co. KG
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+#include <future>
+#include <thread>
+#include <chrono>
+#include <functional>
+
+#include <gtest/gtest.h>
+#include <gmock/gmock.h>
+
+// Test frameworks
+#include "psen_scan_v2/async_barrier.h"
+#include "psen_scan_v2/mock_udp_server.h"
+
+// Software under testing
+#include "psen_scan_v2/scanner_configuration.h"
+#include "psen_scan_v2/scanner.h"
+#include "psen_scan_v2/start_request.h"
+#include "psen_scan_v2/scanner_reply_msg.h"
+
+namespace psen_scan_v2_test
+{
+static const std::string SCANNER_IP_ADDRESS{ "127.0.0.1" };
+static constexpr unsigned short DATA_PORT_OF_SCANNER_DEVICE{ 2000 };
+static constexpr unsigned short CONTROL_PORT_OF_SCANNER_DEVICE{ 3000 };
```

Comment:
Only a minor thing. I would rename to `SCANNER_UDP_PORT_DATA` and `SCANNER_UDP_PORT_CONTROL` in accordance to the other constants.

Reply:
Done.